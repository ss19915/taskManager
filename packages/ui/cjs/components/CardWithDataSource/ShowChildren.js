"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(source, true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(source).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var childWithProps = function childWithProps(child, props, key) {
  if (typeof child === 'string' | typeof child === 'number') {
    return child;
  }

  return _react["default"].cloneElement(child, _objectSpread({
    key: key
  }, props));
};

var ShowChildren = function ShowChildren(props) {
  var children = props.children,
      childrenProps = props.childrenProps;

  if (!children) {
    return null;
  }

  if (Array.isArray(children)) {
    return children.map(function (child, index) {
      return childWithProps(child, childrenProps, index);
    });
  }

  return childWithProps(children, childrenProps);
};

ShowChildren.propTypes = {
  children: _propTypes["default"].node,
  childrenProps: _propTypes["default"].object
};
var _default = ShowChildren;
exports["default"] = _default;