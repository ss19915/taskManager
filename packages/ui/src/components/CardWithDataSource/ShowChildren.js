import React from 'react';
import T from 'prop-types';

const childWithProps = (child, props, key) => {
    if (typeof (child) === 'string' | typeof (child) === 'number') {
        return child;
    }

    return React.cloneElement(child, { key: key, ...props });
}

const ShowChildren = (props) => {
    const {
        children,
        childrenProps
    } = props;

    if (!children) {
        return null;
    }

    if (Array.isArray(children)) {
        return (children.map((child, index) => childWithProps(child, childrenProps, index)));
    }

    return childWithProps(children, childrenProps);
};

ShowChildren.propTypes = {
    children: T.node,
    childrenProps: T.object,
};

export default ShowChildren;