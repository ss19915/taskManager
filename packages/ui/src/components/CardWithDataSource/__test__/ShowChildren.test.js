import React from 'react';
import ShowChildren from '../ShowChildren';

describe('Component: ShowChildren', () => {
    it('Should render correctly with no child', () => {
        const wrapper = shallow(<ShowChildren/>);

        expect(wrapper).toMatchSnapshot();
    });

    it('Should render correctly with one child', () => {
        const A = () => (<p>A</p>)

        const wrapper = shallow(
            <ShowChildren>
                <A/>
            </ShowChildren>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('Should render correctly with more than one child', () => {
        const A = () => (<p>A</p>)
        const B = () => (<p>B</p>)

        const wrapper = shallow(
            <ShowChildren>
                <A/>
                <B/>
            </ShowChildren>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('Should pass props with one child', () => {
        const A = () => (<p>A</p>)
        const childrenProps = {a:1};

        const wrapper = shallow(
            <ShowChildren childrenProps={childrenProps}>
                <A/>
            </ShowChildren>
        );

        expect(wrapper.find('A').props()).toEqual(childrenProps);
    });

    it('Should pass props with all childs', () => {
        const A = () => (<p>A</p>)
        const B = () => (<p>B</p>)
        const childrenProps = {a:1};

        const wrapper = shallow(
            <ShowChildren childrenProps={childrenProps}>
                <A/>
                <B/>
            </ShowChildren>
        );

        expect(wrapper.find('A').props()).toEqual(childrenProps);
        expect(wrapper.find('B').props()).toEqual(childrenProps);
    });
});