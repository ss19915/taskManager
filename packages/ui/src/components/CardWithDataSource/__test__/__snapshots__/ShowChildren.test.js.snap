// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ShowChildren Should render correctly with more than one child 1`] = `
Array [
  <A
    key="0"
  />,
  <B
    key="1"
  />,
]
`;

exports[`Component: ShowChildren Should render correctly with no child 1`] = `""`;

exports[`Component: ShowChildren Should render correctly with one child 1`] = `<A />`;
